clang_combined_time_trace.py

```diff
@@ -0,0 +1,366 @@
+
+            existed_entity.combine(entity)
+        else:
+            self.entities[entity.name] = entity
+
+    def _set_total(self, name, duration):
+        assert self.total[name] == 0, 'Non-unique event {}'.format(name)
+
+        self.total[name] = duration
+
+
+class TotalStatistics(Statistics):
+    def __init__(self):
+        Statistics.__init__(self)
+        self.file_count = 0
+        self.total_trace_file_size = 0
+
+    def process_file_stats(self, file_size, file_statistics):
+        self.file_count += 1
+        self.total_trace_file_size += file_size
+
+        file_statistics.finish()
+
+        for name, time in file_statistics.total.items():
+            self.total[name] += time
+
+        for entity_name, entity in file_statistics.entities.items():
+            if entity_name in self.entities:
+                self.entities[entity_name].combine(entity)
+            else:
+                self.entities[entity_name] = entity
+
+
+class ProgressBar:
+    def __init__(self, total):
+        self.total = total
+        self.current = 0
+        self.prev_len = 0
+
+    def next(self, string):
+        self._clear()
+
+        self.current += 1
+        assert self.current <= self.total, 'Incorrect total task count'
+
+        current_string = '[{}/{}] {}'.format(self.current, self.total, string)
+        self.prev_len = len(current_string)
+        print(current_string, end='', flush=True)
+
+    def _clear(self):
+        import sys
+
+        if self.current == 0:
+            return
+
+        if sys.__stdout__.isatty():
+            print('\r', end='')
+            print(' ' * self.prev_len, end='')
+            print('\r', end='')
+        else:
+            print('')
+
+
+@contextmanager
+def create_progress_bar(total):
+    progress_bar = ProgressBar(total)
+    yield progress_bar
+    print('', flush=True)
+
+
+def find_traces(dir_path):
+    trace_list = []
+    for subpath in dir_path.iterdir():
+        if subpath.is_dir():
+            trace_list.extend(find_traces(subpath))
+        if subpath.is_file() and subpath.parts[-1].endswith('.json'):
+            trace_list.append(subpath)
+    return trace_list
+
+
+def load_trace(trace):
+    trace_stats = TraceStatistics()
+    for trace_event in trace['traceEvents']:
+        trace_stats.load_event(trace_event)
+    return trace_stats
+
+
+def load_traces(progress_bar, trace_list):
+    import os
+    import json
+
+    stats = TotalStatistics()
+
+    for trace in trace_list:
+        progress_bar.next(trace.parts[-1])
+
+        with open(trace) as json_file:
+            trace_stats = load_trace(json.load(json_file))
+
+        stats.process_file_stats(os.path.getsize(trace), trace_stats)
+
+    return stats
+
+
+def format_time(time_mcs):
+    return '{:.2f}s'.format(time_mcs / 1000000)
+
+
+def format_part_time(time_mcs, total_mcs):
+    return '{} ({:.2%})'.format(
+        format_time(time_mcs), float(time_mcs) / total_mcs)
```

Comment:
Nit: `/` in Python3 is "float" by default - no need to convert it explicitly.

Reply:
Fixed